commands/leaderboard.js

```diff
@@ -0,0 +1,41 @@
+var fs = require("fs");
+
+module.exports = {
+  leaderboard: function (message, roster) {
+
+    if(message.content.split(" ").length !== 2) { // we want users to input how many
+      message.channel.send("Please use like so: !leaderboard <positive integer>");
+      return;
+    }
+
+    let stringy = "List of beautiful whales <3\n```";
+    let amount = message.content.split(" ")[1];
+    amount = Math.min(amount, roster.length); //prevents out of index errors
+
+    if (!parseInt(amount)) { //for NaN inputs
+      message.channel.send("Use a positive WHOLE number you twat");
+      return;
+    }
+
+    if (parseInt(amount) < 1) { // for 0 or negatives
+      message.channel.send("Stop it. Use a positive WHOLE number");
+      return;
+    }
+
```

Comment:
Shorter JS/ES6 version of lines 25-38:
```suggestion
message.channel.send(`\`\`\`${roster.sort((a, b) => b.rank - a.rank)
      .map((member,index) => `#${index+1} ${member.name} ${" ".repeat(Math.abs(member.name.length - 13 + (index+1).toString().length))} ${member.rank}`)
      .slice(0,amount)
      .join('\n')}\`\`\``)
```

Reply:
You might have to add a .repeat after the ${index+1} though to make the spacing more consistent again

Reply:
this is perfect. modified a bit and it is all evenly spaced and nice.
thank u kind sur. much smart. many brain.